core/src/themes/ionic.globals.ionic.scss

```diff
@@ -0,0 +1,209 @@
+
+  /* Font sizes */
+  --ionic-font-size-s: #{$ionic-font-size-s};
+  --ionic-font-size-base: #{$ionic-font-size-base};
+  --ionic-font-size-l: #{$ionic-font-size-l};
+
+  /* Font size displays */
+  --ionic-display-s: #{$ionic-display-s};
+  --ionic-display-m: #{$ionic-display-m};
+  --ionic-line-height-display-s: 44px;
+  --ionic-line-height-display-m: 48px;
+
+  /* Headings */
+  --ionic-h1-font-size: #{$ionic-h1-font-size};
+  --ionic-h2-font-size: #{$ionic-h2-font-size};
+  --ionic-h3-font-size: #{$ionic-h3-font-size};
+  --ionic-h4-font-size: #{$ionic-h4-font-size};
+  --ionic-h5-font-size: #{$ionic-h5-font-size};
+  --ionic-h6-font-size: #{$ionic-h6-font-size};
+
+  /* Font Weights */
+  --ionic-font-light: 300;
+  --ionic-font-regular: 400;
+  --ionic-font-medium: 500;
+  --ionic-font-semibold: 600;
+  --ionic-font-bold: 700;
+
+  /* Line heights */
+  --ionic-line-height-xs: 16px;
+  --ionic-line-height-s: 20px;
+  --ionic-line-height-m: 24px;
+  --ionic-line-height-l: 28px;
+  --ionic-line-height-xl: 32px;
+  --ionic-line-height-xxl: 36px;
+
+  /* Letter spacing */
+  --ionic-letter-spacing-none: 0em;
+  --ionic-letter-spacing-s: 0.1em;
+  --ionic-letter-spacing-m: 0.15em;
+}
+
+html {
+  font-family: var(--ionic-font-family);
+  font-size: var(--ionic-font-size-base);
+  font-weight: var(--ionic-font-regular);
+  line-height: var(--ionic-line-height-m);
+}
+
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+  color: var(--ionic-color-base);
+  font-weight: var(--ionic-font-semibold);
+}
+
+h1 {
+  font-size: var(--ionic-h1-font-size);
+  line-height: var(--ionic-line-height-xxl);
+}
+
+h2 {
+  font-size: var(--ionic-h2-font-size);
+  line-height: var(--ionic-line-height-xl);
+}
+
+h3 {
+  font-size: var(--ionic-h3-font-size);
+  line-height: var(--ionic-line-height-l);
+}
+
+h4 {
+  font-size: var(--ionic-h4-font-size);
+  line-height: var(--ionic-line-height-l);
+}
+
+h5 {
+  font-size: var(--ionic-h5-font-size);
+  line-height: var(--ionic-line-height-l);
+}
+
+h6 {
+  font-size: var(--ionic-h6-font-size);
+  line-height: var(--ionic-line-height-s);
+}
+
+/* Utility Classes */
+/* Headings */
+.ionic-heading1,
+.ionic-heading2,
+.ionic-heading3,
+.ionic-heading4,
+.ionic-heading5,
+.ionic-heading6 {
+  color: var(--ionic-color-base);
+  font-weight: var(--ionic-font-semibold);
+}
+
+.ionic-heading1 {
+  font-size: var(--ionic-h1-font-size);
+  line-height: var(--ionic-line-height-xxl);
+}
+
+.ionic-heading2 {
+  font-size: var(--ionic-h2-font-size);
+  line-height: var(--ionic-line-height-xl);
+}
+
+.ionic-heading3 {
+  font-size: var(--ionic-h3-font-size);
+  line-height: var(--ionic-line-height-l);
+}
+
+.ionic-heading4 {
+  font-size: var(--ionic-h4-font-size);
+  line-height: var(--ionic-line-height-l);
+}
+
+.ionic-heading5 {
+  font-size: var(--ionic-h5-font-size);
+  line-height: var(--ionic-line-height-l);
+}
+
+.ionic-heading6 {
+  font-size: var(--ionic-h6-font-size);
+  line-height: var(--ionic-line-height-s);
+}
```

Comment:
These should be grouped with the heading tags since they share the same styles. Ex:

```scss
h1,
.ionic-heading1 {
   // styles
}
```

Reply:
I divided because it's possible to have a css file only with the utility classes. I will change it, but if in the future we decide to have a different place for utility classes, we need to review this.